packages/jni-swig-stub/realm.i

```diff
@@ -326,6 +326,12 @@ bool throw_as_java_exception(JNIEnv *jenv) {
 // Enable passing uint8_t* parameters for realm_config_get_encryption_key and realm_config_set_encryption_key as Byte[]
 %apply int8_t[] {uint8_t *key};
 %apply int8_t[] {uint8_t *out_key};
+%apply int8_t[] {const uint8_t* data};
+
+%typemap(freearg) const uint8_t* data;
+%typemap(out) const uint8_t* data %{
+    $result = SWIG_JavaArrayOutSchar(jenv, (signed char *)result, arg1->size);
+%}
```

Comment:
The `data` correctly assures that the typemap is only applied when the member name is `data`, hence for `realm_binary_t`. But the current clean up of freeing data in https://github.com/realm/realm-kotlin/pull/894/files#diff-c71ff0791782da90014db6f7000bdeea6c883ca9729f17bd1b11041d1e7711faR703 is only done when `realm_binary_t` is embedded in a `realm_value_t`. This means that if we use a `realm_binary_t` outside the scope of a `realm_binary_t` that is properly clean up, then we would leak. For `realm_string_t` I specifically target assignments to `realm_value_t.string` with the typemap, so that it only copied the buffer in that cases and also ment that we could skip the JNI-roundtrip for setting the individual properties of `realm_string_t` by setting the members in C++. 

Seems like the only other usage of `realm_binary_t` is in `realm_convert_with_path` which we don't even use at the moment, so maybe just add an `%ignore realm_convert_with_path` to `realm.i` for now with a comment around this, as there is probably no need to block this PR further just to circle around this. There might be other updates on how we handle this in the near future anyway (https://mongodb.slack.com/archives/C01LVQ0K404/p1656680435874499?thread_ts=1656673194.461559&cid=C01LVQ0K404)

Reply:
I see, thanks for the explanation 🙇 I will add the comment you suggested then.